### ISSUE TITLE
Optional: Add a Cancel Reservation Button

### ISSUE THREAD

**Issue by Anjacodes** [Anjacodes/Hotelzilla-UI#37]
If there is still time in the end, I think another nice feature would be to have a `Cancel Reservation` functionality in the `Reservations` component

**Comment by williamrolando88** [comment]
With cancel reservation you mean:

- [ ] A) Cancel the operation on the client-side
- [ ] B) Fetch all current user reservations and delete one entry from the database.

Choose the one your heart says is the correct one 🤣

**Comment by Anjacodes** [comment]
> With cancel reservation you mean:
> 
>     * [ ]  A) Cancel the operation on the client-side
> 
>     * [x]  B) Fetch all current user reservations and delete one entry from the database.
> 
> 
> Choose the one your heart says is the correct one 🤣

BUT: Will we need a new endpoint for this or does it already exist? Because if so, this might not make the backend team very happy 😆

**Comment by williamrolando88** [comment]
>BUT: Will we need a new endpoint for this or does it already exist? Because if so, this might not make the backend team very happy 😆

Yes, we definitely need a `delete` CRUD operation for that purpose
